element_facemap/facemap_inference.py

```diff
@@ -0,0 +1,478 @@
+    
+@schema
+class FacemapModel(dj.Manual):
+    """Trained Models stored for facial pose inference
+
+    Attributes:
+        model_id(int) : User specified ID associated with a unique model 
+        model_name( varchar(64) ): Name of model, filepath.stem
+    """
+
+    definition = """
+    model_id   : int                    # user assigned ID associated with a unique model
+    ---
+    model_name : varchar(64)            # name of model 
+    model_description: varchar(1000)    # optional model description
+    """
+
+    class BodyPart(dj.Part):
+        """Body parts associated with a given model
+
+        Attributes:
+            body_part ( varchar(32) ): Body part name, (location specfication)
+            body_part_description ( varchar(1000) ): Optional. Longer description."""
+
+        definition = """
+        -> master
+        -> BodyPart
+        """
+
+    class File(dj.Part):
+        """Relative paths of facemap models with respect to facemap_root_data_dir
+
+        Attributes:
+            FacemapModel (foreign key): Facemap model primary key.
+            model_file ( attach ): filepath of facemap model, relative to root data dir
+        """
+
+        definition = """
+        -> master
+        ---
+        model_file: attach            # model file attachment
+        """
+    @classmethod
+    def insert_new_model(cls, model_id: int, model_name: str, model_description: str, full_model_path: str):
+        facemap_model_entry = dict(
+            model_id=model_id, model_name=model_name, model_description=model_description
+        )
+        FacemapModel.insert1(facemap_model_entry)
+
+        body_part_entry = []
+        for bp in BodyPart.fetch('body_part'):
+            body_part_entry.append(dict(model_id=model_id, body_part=bp))
+        
+        file_entry = dict(model_id=model_id, model_file=full_model_path)
+
+        cls.BodyPart.insert(body_part_entry)
+        cls.File.insert1(file_entry)
+
+@schema
+class FacemapPoseEstimationTask(dj.Manual):
+    """Staging table for pairing of video recordings and Facemap parameters before processing.
+
+    Attributes:
+        fbe.VideoRecording (foreign key) : Primary key for VideoRecording table.
+        FacemapModel (foreign key) : Primary key for the facemap model table
+        pose_estimation_output_dir ( varchar(255), optional) : output dir storing the results
+            of pose estimation analysis.
+        task_mode (enum) : Default trigger. 'load' or 'trigger' analysis.
+        bbox (longblob) : Bounding box for cropping the video [x1, x2, y1, y2]. If not set, entire frame is used.
+        task_description ( varchar(128), optional) : Task description.
+    """
+
+    definition = """
+    # Staging table for pairing of recording and Facemap parameters before processing.
+    -> fbe.VideoRecording
+    -> FacemapModel
+    ---
+    pose_estimation_output_dir=''   : varchar(255)  # output dir - stores results of Facemap Pose estimation analysis
```

Comment:
```suggestion
    pose_estimation_output_dir   : varchar(255)  # output dir - stores results of Facemap Pose estimation analysis
```
We should stay away from keeping this optional.